Source/Core/VideoCommon/VertexLoaderManager.h

```diff
@@ -66,7 +69,30 @@ extern u32 g_current_components;
 extern BitSet8 g_main_vat_dirty;
 extern BitSet8 g_preprocess_vat_dirty;
 extern bool g_bases_dirty;  // Main only
-extern u8 g_current_vat;    // Main only
 extern std::array<VertexLoaderBase*, CP_NUM_VAT_REG> g_main_vertex_loaders;
 extern std::array<VertexLoaderBase*, CP_NUM_VAT_REG> g_preprocess_vertex_loaders;
+
+template <bool IsPreprocess = false>
+VertexLoaderBase* RefreshLoader(int vtx_attr_group)
+{
+  const BitSet8& attr_dirty = IsPreprocess ? g_preprocess_vat_dirty : g_main_vat_dirty;
```

Comment:
Bit of a micro-optimization most likely but this could be rewritten to avoid the if check in favor of a `if constexpr`.

```
auto calculate_loader = [](const BitSet8& attr_dirty, const std::array<VertexLoaderBase*, CP_NUM_VAT_REG>& vertex_loaders, int vtx_attr_group)
{
    if (!attr_dirty[vtx_attr_group]) [[likely]]
    {
      return vertex_loaders[vtx_attr_group];
    }
    else [[unlikely]]
    {
      return GetOrCreateLoader<IsPreprocess>(vtx_attr_group);
    }
};

if constexpr (IsPreprocess)
{
  loader = calculate_loader(g_preprocess_vat_dirty, g_preprocess_vertex_loaders, vtx_attr_group);
}
else
{
  loader = calculate_loader(g_main_vat_dirty, g_main_vertex_loaders, vtx_attr_group);
  // Lookup pointers for any vertex arrays.
  UpdateVertexArrayPointers();
}

return loader;
```

Reply:
If what @JosJuice and @Pokechu22 are saying are accurate, this workaround shouldn't be necessary.

Just make the ternary variables `constexpr` and call it a day!

Reply:
What's the point of this? I don't really see what this is supposed to optimize.

Reply:
Yeah, I'd say just mark `attr_dirty` and `vertex_loaders` as `constexpr` instead of adding this lambda. (Lambda functions are scarier from a "will the compiler optimize this or not" perspective than what's going on in the current version of this PR!)

Reply:
@K0bin  - the point of this was to use `if constexpr` instead of your ternarys.  But as discussed, we can just mark the variables as `constexpr` and the compiler will do the right thing.  My bad!